ledger/block/src/transaction/merkle.rs

```diff
@@ -123,22 +141,16 @@ impl<N: Network> Transaction<N> {
             Some(program_checksum) => program_checksum.to_bits_le(),
         };
         // Prepare the leaves.
-        let leaves = deployment
-            .program()
-            .functions()
-            .values()
-            .enumerate()
-            .map(|(index, function)| {
-                // Construct the transaction leaf.
-                Ok(TransactionLeaf::new_deployment(
-                    u16::try_from(index)?,
-                    N::hash_bhp1024(&to_bits_le![header, function.to_bytes_le()?])?,
-                )
-                .to_bits_le())
-            })
-            .collect::<Result<Vec<_>>>()?;
+        let leaves = deployment.program().functions().values().enumerate().map(|(index, function)| {
```

Comment:
Unrelated to these changes, but I believe we can parallelize this and some of the other iterators

Reply:
Considering it will be at most `32`, I think the parallelization will actually slowdown the performance